app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1135,18 +1143,31 @@ private void takePhotoForCard(int type) {
     private void selectImageFromGallery(int type) {
         mRequestedImage = type;
 
-        Intent photoPickerIntent = new Intent(Intent.ACTION_PICK);
-        photoPickerIntent.setType("image/*");
-        Intent contentIntent = new Intent(Intent.ACTION_GET_CONTENT);
-        contentIntent.setType("image/*");
-        Intent chooserIntent = Intent.createChooser(photoPickerIntent, getString(R.string.addFromImage));
-        chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[] { contentIntent });
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            // Use the new photo picker on devices where it is available
+            try {
+                // Registers a photo picker activity launcher in single-select mode.
+                pickMedia.launch(new PickVisualMediaRequest.Builder()
+                        .setMediaType(ActivityResultContracts.PickVisualMedia.ImageOnly.INSTANCE)
+                        .build());
+            } catch (ActivityNotFoundException e) {
+                Toast.makeText(getApplicationContext(), R.string.failedLaunchingPhotoPicker, Toast.LENGTH_LONG).show();
+                Log.e(TAG, "No activity found to handle intent", e);
+            }
+        } else {
+            Intent photoPickerIntent = new Intent(Intent.ACTION_PICK);
+            photoPickerIntent.setType("image/*");
+            Intent contentIntent = new Intent(Intent.ACTION_GET_CONTENT);
+            contentIntent.setType("image/*");
+            Intent chooserIntent = Intent.createChooser(photoPickerIntent, getString(R.string.addFromImage));
+            chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[]{contentIntent});
 
-        try {
-            mPhotoPickerLauncher.launch(chooserIntent);
-        } catch (ActivityNotFoundException e) {
-            Toast.makeText(getApplicationContext(), R.string.failedLaunchingPhotoPicker, Toast.LENGTH_LONG).show();
-            Log.e(TAG, "No activity found to handle intent", e);
+            try {
+                mPhotoPickerLauncher.launch(chooserIntent);
+            } catch (ActivityNotFoundException e) {
+                Toast.makeText(getApplicationContext(), R.string.failedLaunchingPhotoPicker, Toast.LENGTH_LONG).show();
+                Log.e(TAG, "No activity found to handle intent", e);
+            }
```

Comment:
If this version check necessary? Looking at https://developer.android.com/reference/androidx/activity/result/contract/ActivityResultContracts.PickVisualMedia it states: "@[RequiresApi](https://developer.android.com/reference/androidx/annotation/RequiresApi)(value = 19)" and "[Intent.ACTION_OPEN_DOCUMENT](https://developer.android.com/reference/android/content/Intent.html#ACTION_OPEN_DOCUMENT()) is used as a final fallback on all Android API 19 or higher devices.". Given Catima's minSdk is 21, we should be able to always follow this code path.

Reply:
I think is better to check the version and if is lower than Android 11, do it like the old version and in that way make sure the same functionality.